### PR TITLE
Allow to block wares delivery into armor workshop

### PR DESCRIPTION
**Pull request by reyandme**
so that player could focus each on certain task (shields or leather armors)

### CODE REVIEW

File: src/gui/pages_game/KM_GUIGameHouse.pas [hunk 1]
```diff
@@ -891,6 +992,27 @@ procedure TKMGUIGameHouse.House_SchoolUnitChange(Sender: TObject; Shift: TShiftS
 end;
 
 
+{Toggle ware delivery for separate resources (wood, leather) in Armor workshop}
+procedure TKMGUIGameHouse.House_ArmorWSDeliveryToggle(Sender: TObject);
+var I: Byte;
```
Comment (Kromster80) [inline src/gui/pages_game/KM_GUIGameHouse.pas]: I should be Integer in almost all iterations

File: src/gui/pages_game/KM_GUIGameHouse.pas [hunk 2]
```diff
@@ -891,6 +992,27 @@ procedure TKMGUIGameHouse.House_SchoolUnitChange(Sender: TObject; Shift: TShiftS
 end;
 
 
+{Toggle ware delivery for separate resources (wood, leather) in Armor workshop}
+procedure TKMGUIGameHouse.House_ArmorWSDeliveryToggle(Sender: TObject);
+var I: Byte;
+    WareType: TWareType;
+    ArmorWS: TKMHouseArmorWorkshop;
+begin
+  WareType := wt_None;
```
Comment (Kromster80) [inline src/gui/pages_game/KM_GUIGameHouse.pas]: It would be much simpler to assign buttons `Tag := I;` in creation and here just map from `const WORKSHOP_WARE: array [1..2] of TWareType = (wtWood, wtLeather); ..... WareType := WORKSHOP_WARE[Sender.Tag]`

File: src/KM_Controls.pas [hunk 3]
```diff
@@ -350,6 +350,15 @@   TKMButtonFlat = class(TKMControl)
     procedure MouseUp(X,Y: Integer; Shift: TShiftState; Button: TMouseButton); override;
 
     procedure Paint; override;
+
+  end;
+
+
+  {FlatButton}
+  TKMButtonFlat = class(TKMButtonFlatCommon)
```
Comment (Kromster80) [inline src/KM_Controls.pas:358]: Why is this needed? Every control has already OnClicked event that we could use.

**Review by Kromster80 — CHANGES_REQUESTED** [review]

File: src/KM_Controls.pas (hunk 3 above)
Comment (reyandme) [reply src/KM_Controls.pas:358]: TKMWaresRow inherits same behaviour as TKMButtonFlat. Sound, highliting etc. So i extract common parent for both of them. The only difference - painting. 
Do you think its better just to copy paste ?

File: src/gui/pages_game/KM_GUIGameHouse.pas (hunk 2 above)
Comment (reyandme) [reply src/gui/pages_game/KM_GUIGameHouse.pas]: +

**Comment by reyandme** [comment]
@Kromster80 here it is, how it looks like:
https://puu.sh/svnXl/00dca688b3.png
Looks absolutely same as for Storehouse or Barracks

**Comment by reyandme** [comment]
Месяц ожидания. 2й пошел.
Month of waiting. 2nd just have started.

**Comment by Kromster80** [comment]
@lewinjh I don't quite like the function duplication between "ware not allowed"-red-triangle and "house does not allow wares in"-doors-closed-button.
For Barracks/Store it is kind of justified. Workshops are in between - they take multiple wares unliek other houses, yet still use "ware-lines" UI.
What's your opinion on this one?

**Comment by lewinjh** [comment]
@Kromster80 The armour workshop is the only house where this feature makes sense since other houses that take multiple wares require both to make anything. So it's really a special case for barracks, store and armour workshop. It might be adding unnecessary complexity given that you can restrict wood delivery to all armour workshops in ware distribution (so if you plan to not make shields you can set wood to armour workshops to zero).

**Comment by Kromster80** [comment]
@lewinjh so what would be the decision? I'm inclining towards "reject"

**Comment by lewinjh** [comment]
@Kromster80 Yeah me too, it's a special case for a single house and basically just an optimisation to save a few wood if you don't plan to make shields in one of your armoury workshops. I'm ok with rejecting it if you are.
